src/action/common/configure_init_service.rs

```diff
@@ -37,6 +41,44 @@ impl ConfigureInitService {
         init: InitSystem,
         start_daemon: bool,
     ) -> Result<StatefulAction<Self>, ActionError> {
+        match init {
+            #[cfg(target_os = "macos")]
+            InitSystem::Launchd => {
+                // No plan checks, yet
+            },
+            #[cfg(target_os = "linux")]
+            InitSystem::Systemd => {
+                // TODO: once we have a way to communicate interaction between the library and the
+                // cli, interactively ask for permission to remove the file
+
+                // NOTE: Check if the service file already exists...
+                if Path::new(SERVICE_DEST).exists() {
+                    return Err(ActionError::FileExists(PathBuf::from(SERVICE_DEST)));
+                }
+                // NOTE: ...and if there are any overrides in the most well-known places for systemd
+                if Path::new(&format!("{SERVICE_DEST}.d")).exists() {
+                    return Err(ActionError::DirExists(PathBuf::from(format!(
```

Comment:
I think if it already exists, we can check the contents are what we would write, and if so, mark this completed and move on?

Reply:
(assuming this was for the actual service file, not the overrides dir) Well, we wouldn't be writing anything, we're just dropping symlinks...

Reply:
Maybe we can check if it's a symlink going where we want already? If the symlink already existed we don't need to do anything...

Reply:
We could, yeah. Do we care about the case where it's a symlink with multiple layers of indirection? i.e. link to `file1` that links to `file2` that eventually links to the  service file itself? Or do we only read the link once and error otherwise?

Reply:
I opted for the simple "no"